src/components/accessories/patientOPD/patientOPD.tsx

```diff
@@ -65,6 +74,10 @@ const PatientOPD: FunctionComponent = () => {
       t("common.somethingwrong")
   ) as string;
 
+  const open = useMemo(() => {
+    return creationMode ? canCreate : canUpdate;
+  }, [creationMode, canCreate, canUpdate]);
+
```

Comment:
The "open" name is too much misleading, please rename it in something more contextual here and in the other files.